controller/hybridgateway/controller.go

```diff
@@ -129,6 +131,22 @@ func (r *HybridGatewayReconciler[t, tPtr]) Reconcile(ctx context.Context, req ct
 	gvk := obj.GetObjectKind().GroupVersionKind()
 	log.Debug(logger, "Reconciling object", "Group", gvk.Group, "Kind", gvk.Kind)
 
+	// Handle deletion and finalizer cleanup
+	if obj.GetDeletionTimestamp() != nil {
+		return r.handleDeletion(ctx, logger, obj, rootObj)
+	}
+
+	// Ensure finalizer is present
+	if !controllerutil.ContainsFinalizer(obj, routeconst.RouteFinalizer) {
+		log.Debug(logger, "Adding finalizer")
+		controllerutil.AddFinalizer(obj, routeconst.RouteFinalizer)
+		if err := r.Update(ctx, obj); err != nil {
```

Comment:
We should use `Patch` here, instead of Update, to minimize the risks of conflicts